client/components/UI/Form/SelectUserInput/index.jsx

```diff
@@ -67,9 +67,17 @@ export class SelectUserInput extends React.Component {
         });
     }
 
+    getUsersToDisplay(list = []) {
+        if (!this.props.hideInactiveDisabled) {
+            return list;
+        } else {
+            return list.filter((u) => u.is_active && u.is_enabled && !u.needs_activation);
+        }
+    }
+
     render() {
         const {value, users, popupContainer, label, readOnly} = this.props;
-        const userList = this.state.searchText ? this.state.filteredUserList : users;
+        const userList = this.state.searchText ? this.state.filteredUserList : this.getUsersToDisplay(users);
```

Comment:
Don't filter the users in any render function, as multiple renders can occur in a very short period of time. Instead place this filtering in component lifecycles such as `willReceiveProps` or `onChange`